code/modules/atmospherics/machinery/pipes/heat_exchange/he_pipes.dm

```diff
@@ -43,11 +43,15 @@
 
 
 	//heatup/cooldown any mobs buckled to ourselves based on our temperature
-	if(buckled_mob)
+	if(buckled_mobs.len)
 		var/hc = pipe_air.heat_capacity()
-		var/avg_temp = (pipe_air.temperature * hc + buckled_mob.bodytemperature * 3500) / (hc + 3500)
+		var/mob/living/heat_source = buckled_mobs[1]
+		var/avg_temp = (pipe_air.temperature * hc + (heat_source.bodytemperature * buckled_mobs.len) * 3500) / (hc * buckled_mobs.len + 3500) //this is wrong, but w/e
```

Comment:
I'm multiplying the body temperature of the first mob by the number of mobs, to get a guess estimate of the total body temperature of all mobs.

I really didn't want to have two loops, one to gather the total heat and then another to apply it.